python_modules/libraries/dagster-k8s/dagster_k8s/client.py

```diff
@@ -91,6 +93,39 @@ class DagsterK8sJobStatusException(Exception):
 ]
 
 
+class PatchedApiClient(ApiClient):
+    # Forked from ApiClient implementation to pass configuration object down into created model
+    # objects, avoiding lock contention issues. See https://github.com/kubernetes-client/python/issues/2284
+    def __deserialize_model(self, data, klass):
```

Comment:
hmm this forking feels a bit more dicey - do we need to have stricter versioning ? how long has this method been the same in their code base ?